src/System.Net.Security/src/System/Net/Security/TlsCipherSuiteNameParser.ttinclude

```diff
@@ -0,0 +1,584 @@
+﻿<#@ assembly name="System.Core" #>
+<#@ import namespace="System" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#+
+
+// Copy of ssl enum types with values which we are missing
+// We will translate those names to the values we have available
+// We still want to have the source of truth somewhere
+public enum ExchangeAlgorithmType
+{
+    None,
+    Rsa,
+    DiffieHellmanStatic,
+    DiffieHellmanEphermal,
+    ECDiffieHellman,
+    ECDiffieHellmanEphermal,
+    Kerberos5,
+    PSK,
+    SRP,
+    ECCPWD,
+    Any,
+}
+
+public enum CipherAlgorithmType
+{
+    Aes,
+    Aes128,
+    Aes192,
+    Aes256,
+    Des,
+    None,
+    Null,
+    Rc2,
+    Rc4,
+    TripleDes,
+    AesGcm,
+    AesCcm,
+    Aes128Gcm,
+    Aes256Gcm,
+    Aes128Ccm,
+    Aes128Ccm8,
+    Aes256Ccm,
+    Aes256Ccm8,
+    Camellia,
+    Camellia128,
+    Camellia256,
+    Camellia128Gcm,
+    Camellia256Gcm,
+    ChaCha20,
+    ChaCha20Poly1305,
+    Seed,
+    Idea,
+    Aria,
+    Aria128,
+    Aria256,
+    Aria128Gcm,
+    Aria256Gcm,
+}
+
+public enum HashAlgorithmType
+{
+    None,
+    Md5,
+    Sha1,
+    Sha256,
+    Sha384,
+    Sha512,
+    Aead,
+}
+
+internal static class EnumHelpers
+{
+    public static string ToFrameworkName(ExchangeAlgorithmType val)
+    {
+        switch (val)
+        {
+            case ExchangeAlgorithmType.None:
+                return "None";
+            case ExchangeAlgorithmType.Rsa:
+                return "RsaKeyX";
+            case ExchangeAlgorithmType.DiffieHellmanEphermal:
+            case ExchangeAlgorithmType.DiffieHellmanStatic:
+            case ExchangeAlgorithmType.ECDiffieHellman:
+            case ExchangeAlgorithmType.ECDiffieHellmanEphermal:
+                return "DiffieHellman";
+            case ExchangeAlgorithmType.Kerberos5:
+            case ExchangeAlgorithmType.PSK:
+            case ExchangeAlgorithmType.SRP:
+            case ExchangeAlgorithmType.ECCPWD:
+            case ExchangeAlgorithmType.Any:
+                return  "None";
+            default:
+                throw new Exception($"Value `{val}` does not have framework corresponding name.");
```

Comment:
I don't know what the t4 exceptions look like.  Should this mention that it's the ExchangeAlgorithmType resolver?  (Similar to the other one(s))

Reply:
added reference to a file in the exception message and method name and type it refers to